static/app/components/deprecatedAsyncComponent.tsx

```diff
@@ -82,20 +77,11 @@ class DeprecatedAsyncComponent<
     }
   }
 
-  componentDidUpdate(prevProps: P, prevContext: any) {
-    const isRouterInContext = !!prevContext.router;
```

Comment:
I am confused. React docs for legacy context dont state that after React 16, componentDidUpdate no longer receives the context value as the second argument, so your type change here seems correct, but does that mean that this has been broken for quite a while now?

Reply:
`isRouterInContext` was always false, yes, so this code branch can't have worked. My guess is that all subclasses either overwrite `componentDidUpdate` without calling `super` (I've seen some of those) or the components are just wrapped in the `withRouter` HoC, so `location` was always in props and never taken from the non-existing context.

So very likely, this was just dead code.